### PR TITLE
Use Liqwid-Nix/ Use latest Plutarch

### PR DESCRIPTION
**Pull request by SeungheonOh**
- I'm not sure dropping `PEq` instance for `Unsorted` maps in Plutarch is intentional or not. This made some tests broken, so they had to go.
- Need review specifically on `PTryFrom` instance of `PTagged`. It had to give `PSubtype` constraint, but I'm not sure if that is a correct way.
- I made other changes to `connor/liqwid-nix` branches of `plutarch-*` repos. Mostly rebasing them to staging. 
    - We need to get those merged soon, before this PR.
- I think this PR(branch) is off of #39 (Am I right @chfanghr?). 

EDIT(by @chfanghr):
- This PR is indeed based on #39
- According to [this graph](https://gist.github.com/chfanghr/9c73f0f7af6e08f3f651163abaa70018), we will have to merge the following PRs first:

  - [x] https://github.com/Liqwid-Labs/plutarch-numeric/pull/5
  - [x] https://github.com/Liqwid-Labs/plutarch-context-builder/pull/19
  - [x] https://github.com/Liqwid-Labs/plutarch-quickcheck/pull/24

### CODE REVIEW

**Comment by SeungheonOh** [comment]
Hint: Las told me we get orphan instance of `PTryFrom` for that `PTagged`.

**Comment by chfanghr** [comment]
> I think this PR(branch) is off of https://github.com/Liqwid-Labs/liqwid-plutarch-extra/pull/39 (Am I right @chfanghr?).

Yes it is.

**Comment by SeungheonOh** [comment]
Note: Las also told me dropping `PEq` for non sorted map was indeed intentional.

**Comment by chfanghr** [comment]
```hs
type family Helper (a :: PType) (b :: PType) (bi :: PType) :: Bool where
  Helper _ b b = 'False
  Helper a _ bi = PSubtype' a bi

type family PSubtype' (a :: PType) (b :: PType) :: Bool where
  PSubtype' a a = 'True
  PSubtype' a b = Helper a b (PInner b)

type family PSubtype (a :: PType) (b :: PType) :: Constraint where
  PSubtype a b = PSubtype' a b ~ 'True
```

I think this is equal to saying `PInner (PTagged tag underlying) ~ underlying`, so it's fine.

**Comment by SeungheonOh** [comment]
> This instance is an orphan instance
> Don't make it

Well... I guess we can remove it later?

**Comment by chfanghr** [comment]
We need this to make `PDiscrete` work, it seems. 

https://github.com/Liqwid-Labs/plutarch-safe-money/pull/8/commits/dcef736d6e95d46e7b8209e1263304981694b9c4#diff-426bba56be81134157b66f0027e41c7fb24be4a7ccce19fec1e4273dec3f034dR105-R110

File: src/Plutarch/Extra/IsData.hs [hunk 1]
```diff
@@ -105,11 +118,19 @@ gProductFromBuiltinData (BuiltinData (List xs)) = do
...
 
-  @since 1.1.0
--}
+-- | @since 1.4.0
+instance
+    (PDataFields (PInner a), PInner (DerivePDataFieldsViaDataList a) ~ a) =>
+    PDataFields (DerivePDataFieldsViaDataList a)
+    where
+    type PFields (DerivePDataFieldsViaDataList a) = PFields (PInner a)
+    ptoFields x = ptoFields $ pto $ pto x
```
Comment (t1lde) [inline src/Plutarch/Extra/IsData.hs:131]: This should work just fine, but I wonder if you'd be able to use IsWrappedType from SOP.Generic :thinking:

File: src/Plutarch/Extra/IsData.hs [hunk 2]
```diff
@@ -105,11 +118,19 @@ gProductFromBuiltinData (BuiltinData (List xs)) = do
...
-  Unsafe version of 'gProductFromBuiltinData'.
+-- | @since 1.4.0
+newtype DerivePDataFieldsViaDataList (a :: S -> Type) (s :: S)
+    = DerivePDataFieldsViaDataList (a s)
 
-  @since 1.1.0
--}
+-- | @since 1.4.0
+instance
+    (PDataFields (PInner a), PInner (DerivePDataFieldsViaDataList a) ~ a) =>
```
Comment (t1lde) [inline src/Plutarch/Extra/IsData.hs:127]: PInner no longer has the awkward extra type variable :partying_face: !

Comment (chfanghr) [reply src/Plutarch/Extra/IsData.hs:127]: Yeah that's a huge improvement:)

**Review by peter-mlabs — APPROVED** [review]
This LGTM; added to the change log and merging

**Pull request merged**